tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_consolidation_request.py

```diff
@@ -887,6 +887,38 @@ def test_incorrect_target_with_bls_credential(spec, state):
     )
 
 
+@with_electra_and_later
+@with_presets([MINIMAL], "need sufficient consolidation churn limit")
+@with_custom_state(
+    balances_fn=scaled_churn_balances_exceed_activation_exit_churn_limit,
+    threshold_fn=default_activation_threshold,
+)
+@spec_test
+@single_phase
+def test_incorrect_source_with_bls_credential(spec, state):
+    # move state forward SHARD_COMMITTEE_PERIOD epochs to allow for consolidation
+    state.slot += spec.config.SHARD_COMMITTEE_PERIOD * spec.SLOTS_PER_EPOCH
+
+    current_epoch = spec.get_current_epoch(state)
+    source_index = spec.get_active_validator_indices(state, current_epoch)[0]
+    target_index = spec.get_active_validator_indices(state, current_epoch)[1]
+    set_compounding_withdrawal_credential_with_balance(spec, state, target_index)
+
+    # It's not possible to control this field, but if you were somehow able to control
+    # the address which shared the same bytes as the validator's BLS pubkey.
+    source_address = state.validators[source_index].withdrawal_credentials[-20:]
```

Comment:
For a more realistic test vector, you could generate an eth1 address instead of fetching it from the existing `withdrawal_credentials`. A random address would probably be more robust, but this seems to be how things are usually written in the codebase:

```suggestion
    assert state.validators[source_index].withdrawal_credentials[-20:] == spec.BLS_WITHDRAWAL_PREFIX
    source_address = b'\x22' * 20
    state.validators[source_index].withdrawal_credentials[-20:] = source_address
```

Reply:
Also asserting that the `withdrawal_credentials` are indeed BLS ones, unless it's a hidden assumption of the given state that I'm not aware of.

Reply:
Yes, checking the credentials type is a good idea 👍 will do. The slicing in your suggestion is wrong though.

And for the other thing. I agree that your suggestion is more realistic, but the withdrawal credentials bytes are immutable, so I don't believe what you suggested would actually work. The current way is a lot simpler.

Reply:
Oops! I meant `[0:1]`, but you get the idea :)
Sounds good to me. I was not aware of that.

Reply:
Haha no worries. Yes I got the idea 😄 thank you 🙏